src/gui_common/charts/line/LineChart.cs

```diff
@@ -0,0 +1,591 @@
+            return;
+        }
+
+        ClearChart();
+
+        var dataSetCount = 0;
+
+        foreach (var data in DataSets)
+        {
+            dataSetCount++;
+
+            // Hide the rest of the dataset if the amount is more than max number to be displayed
+            if (dataSetCount > MaxDisplayedDataSet)
+                UpdateDataSetVisibility(data.Key, false);
+
+            foreach (var point in data.Value.DataPoints)
+            {
+                // Find out value boundaries
+                MaxValues = new Vector2(Mathf.Max(point.Value.x, MaxValues.x), Mathf.Max(point.Value.y, MaxValues.y));
+                MinValues = new Vector2(Mathf.Min(point.Value.x, MinValues.x), Mathf.Min(point.Value.y, MinValues.y));
+
+                // Create tooltip for the point markers
+                var toolTip = ToolTipHelper.CreateDefaultToolTip();
+
+                toolTip.DisplayName = data.Key + point.Value;
+                toolTip.Description = $"{point.Value.x} {XAxisName}\n{point.Value.y} {YAxisName}";
+                toolTip.DisplayDelay = 0;
+
+                ToolTipHelper.RegisterToolTipForControl(point, toolTipCallbacks, toolTip);
+                ToolTipManager.Instance.AddToolTip(toolTip, "chartMarkers" + data.Key);
+            }
+        }
+
+        // Populate the rows
+        for (int i = 1; i < XAxisTicks; i++)
+        {
+            var label = new Label
+            {
+                SizeFlagsHorizontal = (int)SizeFlags.ExpandFill,
+                Align = Label.AlignEnum.Right,
+            };
+
+            label.Text = Mathf.Round(i * (MaxValues.x - MinValues.x) /
+                (XAxisTicks - 1) + MinValues.x).ToString(CultureInfo.CurrentCulture);
+
+            horizontalLabelsContainer.AddChild(label);
+        }
+
+        // Populate the columns (in reverse order)
+        for (int i = YAxisTicks; i-- > 0;)
+        {
+            var label = new Label
+            {
+                SizeFlagsVertical = (int)SizeFlags.ExpandFill,
+                Align = Label.AlignEnum.Center,
+                Valign = Label.VAlign.Bottom,
+            };
+
+            // Don't set the text for the 0 scale label, since there's already one on the x axis.
+            // Add the label nevertheless so the vertical ticks spacings will remain consistent
+            if (i > 0)
+            {
+                label.Text = Mathf.Round(i * (MaxValues.y - MinValues.y) /
+                    (YAxisTicks - 1) + MinValues.y).ToString(CultureInfo.CurrentCulture);
+            }
+
+            verticalLabelsContainer.AddChild(label);
+        }
+
+        if (!string.IsNullOrEmpty(legendTitle))
+            CreateLegend(legendTitle);
+
+        drawArea.Update();
+    }
+
+    public void ClearChart()
+    {
+        toolTipCallbacks.Clear();
+
+        foreach (var data in DataSets)
+        {
+            ToolTipManager.Instance.ClearToolTips("chartMarkers" + data.Key);
```

Comment:
Won't this conflict if multiple charts exist at once? I think that you'll need to add a unique counting up number for each chart instance to make the names unique.

Reply:
What if just adding like a chart name field so the chart could have specific names for each instance (assuming they would be named differently), and then simply use it on the tooltip group names? Maybe they can be used for other purposes in the future too.

Reply:
That would work as well. I suppose it isn't that much work to require the GUI designers to set an unique name on all charts that can exist concurrently. The chart _Ready should throw an error if the name is unset to guard against someone forgetting to set a name.